### ISSUE TITLE
IE 11 compatibility

### ISSUE THREAD

**Issue by jbranigan** [azavea/adb-mongolia-nua#19]
There are at least two issues on IE 11 (running on Windows 7):
- The legends on the map do not render
- The report does not load - not a blank page, just nothing happens

**Comment by rmartz** [comment]
The error message for the Data page is `ReferenceError: 'Symbol' is undefined`, created in `getAllFields()`. The problem seems to be from this compiled JS block:

```
          for (var _iterator5 = Object.keys(Config.mapSections)[Symbol.iterator](), _step5; !(_iteratorNormalCompletion5 = (_step5 = _iterator5.next()).done); _iteratorNormalCompletion5 = true) {
            var label = _step5.value;

            loadSectionFields(fields, Config.mapSections[label], 'visualizations');
          }
```

(Symbol isn't from the ES6 source, it's introduced as part of the compilation process)

I'll investigate what the work around for this is.

**Comment by rmartz** [comment]
The missing legends appears on https://adbmongolia.carto.com as well, seems to be a style issue:
<img width="925" alt="screen shot 2016-08-29 at 12 52 50 pm" src="https://cloud.githubusercontent.com/assets/1032849/18059614/c2ee5318-6de7-11e6-95c0-2e270dad4542.png">

The legend pieces are "there", the issue is the color graduations bar isn't given any colors. Not sure why that might be.

**Comment by CloudNiner** [comment]
re: Symbol: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol

No support in IE, we'll need a polyfill.

Maybe this is helpful?
http://stackoverflow.com/questions/33828840/symbol-is-undefined-in-ie-after-using-babel

**Comment by rmartz** [comment]
Yup, I found that StackOverflow this morning and it seemed to be exactly what we were running into.

The CartoDB map legends is a bit trickier. This Github issue seems to describe what we're facing: https://github.com/CartoDB/cartodb/issues/3068. One option would be to explicitly set the styles elsewhere to bypass IE11's unusual behavior with element-defined styles, but different maps can have different colors which is... problematic. We may be able to change the legend styling on the CartoDB side to include a CSS style that has no impact (Like `color: inherit;`), but that would require changing the actual CartoDB configuration which I'm a bit wary of.

**Comment by rmartz** [comment]
The part that's most interesting about this to me isn't that the styles vanish, which is weird, but that elements around the list change completely. This is what the code is supposed to show:
<img width="528" alt="screen shot 2016-08-29 at 4 56 45 pm" src="https://cloud.githubusercontent.com/assets/1032849/18067168/9db9484c-6e09-11e6-8564-1053913defa6.png">

And this is what we get in IE11:
![screen shot 2016-08-29 at 4 55 34 pm](https://cloud.githubusercontent.com/assets/1032849/18067195/b5c61636-6e09-11e6-92ed-7e2feea12ca4.png)

Styles for `quartile` and `colors` and missing, sure, but the styles for `graph leg` are _different_. Something changed them. Which means we either have a ghost writing CSS to mess with us, or there's some logic processing this that's doing something undesired. I checked the payload that we're getting from Carto and it includes the raw, unadulterated styles, and I can't find any hint of this being a larger IE11 bug, so it very likely must be something on the client side that's doing this.

**Comment by designmatty** [comment]
I'll take a look at the legend issue

**Comment by rmartz** [comment]
We might have an issue with the legend that needs fixing, in addition to the IE11 problem. I'm mentioning it here because the fix for this issue may also fix the IE11 problem, depending on what route we take. In any case it may not be worth fixing the IE11 problem until we decide how to correct misalignment like this:
<img width="313" alt="screen shot 2016-09-01 at 4 55 44 pm" src="https://cloud.githubusercontent.com/assets/1032849/18184008/f74eb8a8-7064-11e6-8891-7e724d632380.png">
(This was taken in an IE11 VM when testing that the fake style fix worked for that visualization)

The legend is supposed to render like this:
<img width="299" alt="screen shot 2016-09-01 at 4 56 38 pm" src="https://cloud.githubusercontent.com/assets/1032849/18184032/17bc7d1e-7065-11e6-8d39-be4e823aca0a.png">

**Comment by jbranigan** [comment]
@designmatty - do you think this is something you might be able to help with?

**Comment by designmatty** [comment]
@jbranigan I think this is breaking due to the font-family not loading. The font-family was never actually called in the head and is currently only working if you have lato installed locally. The easiest solution will be to load the fonts up. I'm in the process of doing cleanup on the app so I'm going to test this theory and let you know how it works out.

The potentially best/longer solution would be to remove the `&nbsp;` from between each increment. Then wrap each increment in a `<span>` and then using some css to make the elements display like they're in a table (the exact same way the color breaks are created).

**Comment by designmatty** [comment]
This is what it looks like when the font loads correctly in ie11 (ignore the colors)

<img width="260" alt="screen shot 2016-09-08 at 5 00 07 pm" src="https://cloud.githubusercontent.com/assets/1928955/18366720/d83b4bb2-75e5-11e6-98bb-1dc8f6b3217b.png">

**Comment by designmatty** [comment]
<img width="321" alt="screen shot 2016-09-08 at 5 01 40 pm" src="https://cloud.githubusercontent.com/assets/1928955/18366743/f11b4380-75e5-11e6-9fa5-a90d008ccdc1.png">

**Comment by jbranigan** [comment]
That's looking good to me. @rmartz , what do you think?

**Comment by rmartz** [comment]
Looks fine to me :)
